plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -125,8 +128,29 @@ public BigQueryMetadata(BigQueryClient bigQueryClient, BigQueryConfig config)
     public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName schemaTableName)
     {
         log.debug("getTableHandle(session=%s, schemaTableName=%s)", session, schemaTableName);
-        String remoteSchemaName = bigQueryClient.toRemoteDatasetName(projectId, schemaTableName.getSchemaName());
-        String remoteTableName = bigQueryClient.toRemoteTableName(projectId, remoteSchemaName, schemaTableName.getTableName());
+        String remoteSchemaName = bigQueryClient.toRemoteDataset(projectId, schemaTableName.getSchemaName())
+                .map(RemoteDatabaseObject::getOnlyRemoteName)
+                .orElse(schemaTableName.getSchemaName());
```

Comment:
why not throw `Table [%s.%s] was not found` here?

Reply:
To keep it consistent with BaseJdbcClient.

But I think failing early is useful here to avoid the cost of the remote API calls while JDBC is cheaper. I'll change.

Reply:
No, turns out we cannot do that. A lot of code in the engine depends on a `null` value being returned `getTableHandle` and performing a different action in those cases.